pkg/registry/image.go

```diff
@@ -0,0 +1,60 @@
+package registry
+
+import (
+	"fmt"
+
+	"github.com/docker/distribution/reference"
+	digest "github.com/opencontainers/go-digest"
+)
+
+// Image holds information about an image.
+type Image struct {
+	Domain string
+	Path   string
+	Tag    string
+	Digest digest.Digest
+	named  reference.Named
+}
+
+// String returns the string representation of an image.
+func (i Image) String() string {
+	return i.named.String()
+}
+
+// Reference returns either the digest if it is non-empty or the tag for the image.
+func (i Image) Reference() string {
+	if len(i.Digest.String()) > 1 {
+		return i.Digest.String()
+	}
+
+	return i.Tag
+}
+
+// ParseImage returns an Image struct with all the values filled in for a given image.
+func ParseImage(image string) (Image, error) {
```

Comment:
what does the input look like?

Reply:
like `localhost:5000/nginx:latest`, `nginx:perl`, `harbor.devops.kubesphere.local:30280/library/busybox:1.28.4`

Reply:
add more comments to clarify

Reply:
ok